src/cli/java/org/commcare/util/screen/SyncScreen.java

```diff
@@ -0,0 +1,133 @@
+package org.commcare.util.screen;
+
+import org.commcare.modern.session.SessionWrapper;
+import org.commcare.session.CommCareSession;
+import org.commcare.suite.model.Entry;
+import org.commcare.suite.model.PostRequest;
+import org.commcare.suite.model.RemoteRequestEntry;
+import org.commcare.util.cli.ApplicationHost;
+
+import java.io.IOException;
+import java.io.PrintStream;
+import java.util.Hashtable;
+
+import okhttp3.Credentials;
+import okhttp3.HttpUrl;
+import okhttp3.MultipartBody;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.RequestBody;
+import okhttp3.Response;
+
+/**
+ * Screen to make a sync request to HQ after a case claim. Unlike other all other screens,
+ * SyncScreen does not take input - it simply processes the PostRequest object, makes the request,
+ * and displays the result (if necessary)
+ */
+public class SyncScreen extends Screen {
+
+    protected SessionWrapper sessionWrapper;
+    private String username;
+    private String password;
+    private PrintStream printStream;
+    private boolean syncSuccessful;
+
+    public SyncScreen(String username, String password, PrintStream printStream) {
+        super();
+        this.username = username;
+        this.password = password;
+        this.printStream = printStream;
+    }
+
+    @Override
+    public void init (SessionWrapper sessionWrapper) throws CommCareSessionException {
+        this.sessionWrapper = sessionWrapper;
+        parseMakeRequest();
+    }
+
+    private void parseMakeRequest() throws CommCareSessionException {
+        String command = sessionWrapper.getCommand();
+        Entry commandEntry = sessionWrapper.getPlatform().getEntry(command);
+        if (commandEntry instanceof RemoteRequestEntry) {
+            PostRequest syncPost = ((RemoteRequestEntry)commandEntry).getPostRequest();
+            Response response = makeSyncRequest(syncPost);
+            if (response == null || !response.isSuccessful()) {
+                printStream.println(String.format("Sync failed with response %s", response));
+                printStream.println("Press 'enter' to retry.");
+                return;
+            }
+            syncSuccessful = true;
+            ApplicationHost.restoreUserToSandbox(sessionWrapper.getSandbox(), sessionWrapper, sessionWrapper.getPlatform(), username, password);
+            printStream.println(String.format("Sync successful with response %s", response));
+            printStream.println("Press 'enter' to continue.");
+        } else {
+            // expected a sync entry; clear session and show vague 'session error' message to user
+            throw new CommCareSessionException("Initialized sync request while not on sync screen");
```

Comment:
It looks like issues with the I/O process around the response are now not re-tryable again and kill your session in the event of network issues. 

Can your revisit making the I/O part of this not a binary process?

Also: The primary purpose of the CLI is Debugging, so swallowing issues with vague errors should be our lazt line of defense. We should be as intentional as possible here to fail at the right step to retain the roper context. It seems after your last commit we're actually less likely to output a useful error message than we were before...

Reply:
@ctsims I wrote this to make it clearer but I believe that retrying for `IOException`s worked previously since `makeRequest` would return null. I've confirmed that retrying works by manually throwing an `IOException` from the method as well as turning off my internet. 

Good call on the error printing